prov/shm/src/smr_ep.c

```diff
@@ -954,14 +809,14 @@ static int smr_ep_close(struct fid *fid)
 		free(ep->sock_info);
 	}
 
+	if (ep->util_ep.ep_fid.msg != &smr_no_recv_msg_ops)
```

Comment:
Consider shm is used as both owner and peer. It seems srx_ctx is only initiated when ep is enabled. But here `ep->util_ep.ep_fid.msg` is set in fi_endpoint. Will it cause a segfault if application do `fi_endpoint` and then `fi_close(ep_fid)`? I catch a similar issue in one efa unit test that has such corner case.